packages/apps/dashboard/server/src/modules/details/details.service.ts

```diff
@@ -149,40 +155,62 @@ export class DetailsService {
     return result;
   }
 
-  public async getLeadersByChainId(
-    chainId?: ChainId,
-    take?: number,
+  public async getLeaders(
+    chainId: ChainId,
+    { orderBy, orderDirection, first }: GetLeadersPaginationOptions = {},
   ): Promise<LeaderDto[]> {
-    const chainIds = !chainId
-      ? await this.networksService.getOperatingNetworks()
-      : [chainId];
+    const filter = this.createLeadersFilter(
+      chainId,
+      orderBy,
+      orderDirection,
+      first,
+    );
+
+    const [rawLeaders, reputations] = await Promise.all([
+      OperatorUtils.getLeaders(filter),
+      this.fetchReputations(
+        chainId,
+        orderBy,
+        orderDirection,
+        orderBy !== LeadersOrderBy.REPUTATION && first,
```

Comment:
Probably it should be vice-versa:
```suggestion
        orderBy === LeadersOrderBy.REPUTATION ? first : undefined,
```

so:
- when `orderBy` **_is "by reputation"_**, you get first `N` and then cross them with `getLeaders` response
- when `orderBy` **_is  any leader prop_** (i.e. not "by reputation") - we take all reputations to cross the with N leaders